collector/factory.go

```diff
@@ -0,0 +1,57 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package collector // import "go.opentelemetry.io/ebpf-profiler/collector"
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/xconsumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/xreceiver"
+
+	"go.opentelemetry.io/ebpf-profiler/collector/internal"
+	"go.opentelemetry.io/ebpf-profiler/internal/controller"
+)
+
+var (
+	typeStr = component.MustNewType("otelreceiver")
+
+	errInvalidConfig = errors.New("invalid config")
+)
+
+// NewFactory creates a factory for the receiver.
+func NewFactory() receiver.Factory {
+	return xreceiver.NewFactory(
+		typeStr,
+		defaultConfig,
+		xreceiver.WithProfiles(createProfilesReceiver, component.StabilityLevelAlpha))
+}
+
+func createProfilesReceiver(
+	_ context.Context,
+	_ receiver.Settings, //nolint:gocritic // we must respect the collector API
```

Comment:
How does [FactoryOption](https://pkg.go.dev/go.opentelemetry.io/collector/receiver/xreceiver#FactoryOption) from [xreceiver.WithProfiles](https://pkg.go.dev/go.opentelemetry.io/collector/receiver/xreceiver#WithProfiles) force the function signature containing `context.Context` and `receiver.Settings`?
```suggestion
```

Reply:
I got lost in APIs ...